### DISCUSSION TITLE
Provisioning via git repositories.

### DISCUSSION THREAD

**Discussion by Breee** [grafana/grafana#72005] [General]
## Current State 
Currently we are able to provision:
- Via provisioning feature as mounted folder. (https://grafana.com/docs/grafana/latest/administration/provisioning) 
- Using the k8s operator and provision via configmaps. 

## Limitations
- When using the helm-chart, In k8s we have a limit for the size of configmaps, i.e. not all dashboards can be provisioned if they exceed a certain size. Due to the nature of how grafana dashboards are stored, they quickly exceed this limit. 
- The provisioning via a mounted folder requires extra tweaking and a sidecar that fetches provisioning data, e.g. via git pull 

## Idea
- Create a provisioning provider that periodically fetches a git repository and update (1) dashboards, (2) datasources, (3) plugins and (4)  alerts. 

## Questions: 
- Does https://github.com/grafana/grafana/issues/52916  acually solve all of our issues?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82620. Feel free to continue the discussion around this there. Thank you!
